### DISCUSSION TITLE
VectorStore Filter Behavior Question

### DISCUSSION THREAD

**Discussion by dccrain** [microsoft/semantic-kernel#10016] [Q&A] [answered]
I am trying to figure out how the VectorStore VectorSearchFilter() works. 
I am storing a list of Tags on records that I wanted to filter using logical OR, ie. return any record that meets at least one of these conditions. So far, the behavior seems to be that filters are logical ANDs.

Is there any way to accomplish what I want without making separate searches for each condition?

**Comment by sophialagerkranspandey** [comment]
Thanks for your feedback. We have this on our backlog to integrate into Semantic Kernel. I'm looping in @westey-m to expand further.

**Comment by westey-m** [comment, marked as answer]
@dccrain, yes we currently only support AND and two basic clauses, EqualTo and AnyTagEqualTo.
We have a work item to expand this with support for OR and NOT: #9398
